mesonbuild/utils/universal.py

```diff
@@ -979,18 +986,74 @@ def default_libdir() -> str:
 
 def default_libexecdir() -> str:
     # There is no way to auto-detect this, so it must be set at build time
+    if is_haiku():
+        return 'lib'
     return 'libexec'
 
 
 def default_prefix() -> str:
-    return 'c:/' if is_windows() else '/usr/local'
+    if is_windows():
+        return 'c:/'
+    if is_haiku():
+        return '/boot/system/non-packaged'
+    return '/usr/local'
+
+
+def default_datadir() -> str:
+    if is_haiku():
+        return 'data'
+    return 'share'
+
+
+def default_includedir() -> str:
+    if is_haiku():
+        return 'develop/headers'
+    return 'include'
+
+
+def default_infodir() -> str:
+    if is_haiku():
+        return 'documentation/info'
+    return 'share/info'
+
+
+def default_localedir() -> str:
+    if is_haiku():
+        return 'data/locale'
+    return 'share/locale'
+
+
+def default_mandir() -> str:
+    if is_haiku():
+        return 'documentation/man'
+    return 'share/man'
+
+
+def default_sbindir() -> str:
+    if is_haiku():
+        return 'bin'
+    return 'sbin'
+
+
+def default_sysconfdir() -> str:
+    if is_haiku():
+        return 'settings'
+    return 'etc'
 
 
 def get_library_dirs() -> T.List[str]:
     if is_windows():
         return ['C:/mingw/lib'] # TODO: get programmatically
     if is_osx():
         return ['/usr/lib'] # TODO: get programmatically
+    if is_haiku():
+        # TODO: get programmatically
```

Comment:
What would it take to get this programmatically in Haiku? Not really sure what the best way would be on other systems.

Reply:
It can be acquired with `findpaths` command:
```
> findpaths B_FIND_PATH_LIB_DIRECTORY
/boot/home/config/non-packaged/lib
/boot/home/config/lib
/boot/system/non-packaged/lib
/boot/system/lib
```

Reply:
I would recommend just implementing it now instead of leaving it for a TODO.

Reply:
Meson usually gets the library dirs by asking the compiler to report them.

This function right here is dead code. I'm about to remove it entirely.

Reply:
#11313

Reply:
How this function is supposed to be used? Does it mean runtime libs or compiler libs? If it is dead code, should I undo changes for it?

Reply:
It refers to the paths that the compiler can find libraries at, when using `-lfoo` etc.

e.g. this is used to implement `cc.find_library()`.

Reply:
If so, it should be developments paths. Still not sure should I keep it considering https://github.com/mesonbuild/meson/pull/11313.

```
> findpaths B_FIND_PATH_DEVELOP_LIB_DIRECTORY
/boot/home/config/non-packaged/develop/lib
/boot/home/config/develop/lib
/boot/system/non-packaged/develop/lib
/boot/system/develop/lib
```

Reply:
@X547 I would just wait for that PR to go in and then rebase. I bet it will be in later today if I had to guess.

Reply:
Removed in upstream.